sympy/combinatorics/perm_groups.py

```diff
@@ -3391,6 +3401,113 @@ def transitivity_degree(self):
         else:
             return self._transitivity_degree
 
+    def presentation(G):
+        '''
+        Return an `FpGroup` presentation of the group
+
+        '''
+        from sympy.combinatorics.fp_groups import FpGroup
+        from sympy.combinatorics.coset_table import CosetTable
+        from sympy.combinatorics.free_groups import free_group
+        from sympy.combinatorics.homomorphisms import homomorphism
+        from itertools import product
+
+        def _factor_group_by_rels(G, rels):
+            if isinstance(G, FpGroup):
+                return FpGroup(G.free_group, list(uniq(
+                                    g for g in G.relators + rels)))
+            return FpGroup(G, rels)
+
+        len_g = len(G.generators)
+
+        if len_g == 1:
+            order = G.generators[0].order()
+            # handle the trivial group
+            if order == 1:
+                return free_group([])[0]
+            F, x = free_group('x')
+            return FpGroup(F, [x**order])
+
+        half_gens = G.generators[0:(len_g+1)//2]
+        H = PermutationGroup(half_gens)
```

Comment:
Should H be made the trivial subgroup when `len_g` is small instead of repeatedly dividing by 2?